CMakeLists.txt

```diff
@@ -410,7 +410,7 @@ include(CMake/Testlibstdc++.cmake)
 if(QMC_CXX_STANDARD GREATER_EQUAL 17)
   include(CMake/TestCxx17Library.cmake)
 else()
-  include(CMake/TestCxx14Library.cmake)
+  message(FATAL_ERROR "QMCPACK requires compiler support of c++17 or greater.")
```

Comment:
This function is not responsible for requesting C++17. No change here. If C++14 is needed, it should be allowed. That is why the top level only issues a warning.

Reply:
The point is to stop writing workaround code in c++14 and use the simpler semantics of c++17.  So it should be a fatal error to request less than 17 for the standard.

Reply:
I think we still want to give a warning for higher standard numbers like we have in the past.

Reply:
Lets make this even easier. Change the top-level CMakeLists.txt warning message to an error. We can then remove the 14 check in Testliststdc++.cmake

Reply:
We don't live in a perfect world... The warning is clear, it is on your own. But we should not add even more hurdles when struggling with C++14 need.

Reply:
"sb" ? (typo)

Reply:
somebody

Reply:
Easy decision for me. Require C++17 and abort if not there. The simplicity is preferred and we have determined that the required support is there.

Reply:
If I found an C++ issue, it is reasonable to go back and try c++14.

Reply:
I am confident that Peter or Alfredo will put some C++17 in an always compiled part of the code within a week of this being merged.

Reply:
If you only had c++11 the failure of the TestCxx14Library.cmake that was here was fatal.

Reply:
Then test all major compilers and then we can decide.

Reply:
That is indeed what we are doing.

Reply:
https://github.com/QMCPACK/qmcpack/issues/2790
this is from last year. Some compiler and libraries are out-dated as of today.
Some new member can be oneAPI 2021.3, NVHPC (pick a version with C++17 support)

Reply:
NVHPC since 20.7 has supported c++17
I think that was the first release

Reply:
But I don't know how to really support NVHPC without cmake CUDA as a language

Reply:
NVHPC should work without CUDA. Imagine you hit a bug, it may caused by the CPU part.
NVHPC right now should work with the current CUDA setting.

Reply:
We drive NVHPC just like gcc. Nothing to worry about for this PR.

Reply:
So is this resolved?  Now we just check for a good c++17 library.